pycbc/psd/__init__.py

```diff
@@ -522,26 +522,43 @@ def associate_psds_to_segments(opt, fd_segments, gwstrain, flen, delta_f, flow,
         that precision. If 'double' the PSD will be converted to float64, if
         not already in that precision.
     """
-    psds_and_times = generate_overlapping_psds(opt, gwstrain, flen, delta_f,
-                                       flow, dyn_range_factor=dyn_range_factor,
-                                       precision=precision)
-
-    for fd_segment in fd_segments:
-        best_psd = None
-        psd_overlap = 0
-        inp_seg = segments.segment(fd_segment.seg_slice.start,
-                                   fd_segment.seg_slice.stop)
-        for start_idx, end_idx, psd in psds_and_times:
-            psd_seg = segments.segment(start_idx, end_idx)
-            if psd_seg.intersects(inp_seg):
-                curr_overlap = abs(inp_seg & psd_seg)
-                if curr_overlap > psd_overlap:
-                    psd_overlap = curr_overlap
-                    best_psd = psd
-        if best_psd is None:
-            err_msg = "No PSDs found intersecting segment!"
-            raise ValueError(err_msg)
-        fd_segment.psd = best_psd
+    if opt.use_precomputed_psd and opt.precomputed_psd_file:
```

Comment:
It feels like we only need `opt.precomputed_psd_file` and if this is given we assume that it will be used. (so remove the `--use-precomputed-psd` option).

Reply:
Yes, I agree.